app/Http/Backend/Controllers/ProductsController.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace App\Http\Backend\Controllers;
+
+use App\Models\Product;
+use Illuminate\Http\Request;
+
+use App\Http\Backend\Requests;
+use App\http\Backend\Requests\CreateProductsRequest;
+use App\http\Backend\Requests\UpdateProductsRequest;
+use App\Http\Backend\Controllers\Controller;
+use DB;
+use App\Models\Category;
+use Flash;
+
+class ProductsController extends Controller
+{
+    /**
+     * Display a listing of the products.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $products = Product::with('category')->get();
+        return view('backend.products.index', compact('products'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $title = 'Create new products';
+        $categories = Category::pluck('name', 'id');
+        return view('backend.products.create', compact('title', 'categories'));
+    }
+
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request create
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CreateProductsRequest $request)
+    {
+        $messages = [
+            'success' => 'Create new products success.',
+            'error'   => 'Create new products failed.'
+        ];
+        $data = $request->all();
+        if ($request->hasFile('image')) {
+            $image = $request->file('image');
+            $data['image'] = Product::upload($image);
+        }
+        Product::create($data);
+        Flash::success($messages['success']);
+        return redirect('/admin/products');
```

Comment:
it's not good to use absolute url path to redirect, use route name is better way